### PR TITLE
feat: set up pm2 ecosystem

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Initially I was following the path in https://github.com/freeCodeCamp/demo-projects/pull/50, but I diverged a bit.

The idea here is to keep the individual apps as independent as possible.  All they need to do is provide a package.json with an `npm start` script and have their own .env file.  That's it.  That and live in apps/name-of-app, and appear in the list of workspaces, of course.

What the apps can't do is manage their own processes, so I had to remove nodemon. If they do, pm2 can't tell when they've crashed, since nodemon keeps on going.

With that caveat, it should be trivial to add the RWD projects, regardless of how you want to serve them, @moT01.  As long as it can be invoked by `npm start` it should be fine.  Also, by keeping them independent, if we ever find ourselves needing to scale, it should be possible with the minimum amount of fuss.

@raisedadead what do you reckon?

### Remaining tasks

- [x] create single update script?
- [x] update docs with workflow to update all apps
- [x] update docs with workflow to update a single app
- [x] confirm that npm restart picks up any changes to the .env files

### CODE REVIEW

File: apps/personal-library/db/BookModel.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  commentcount: Number,
-});
-
-const BookModel = Mongoose.model("book", BookSchema);
-
-module.exports = {
-  BookSchema,
-  BookModel,
-};
+const Mongoose = require("mongoose");
```
Comment (ojeytonwilliams) [inline apps/personal-library/db/BookModel.js:1]: Not sure why GitHub took exception to this one.  CRLF, maybe?  Either way, it's unchanged if you ignore spaces.

**Comment by raisedadead** [comment]
Hi @ojeytonwilliams 

I did not look into the changes, just skimmed over the main `package.json`.

I am fine with any workflow as long as someone like me (primarily responsible for housekeeping) can do a consistent update of the tooling, deps, etc. in < 15 mins on this whole project. So any tooling, scripts, etc. in that regard is highly appreciated. 

At some point in the future, I will end up deploying these apps via `nomad` (workload orchestration).

**Comment by ojeytonwilliams** [comment]
Hey @raisedadead 👋 I didn't change much in the package.json from how you had it in your PR.  There are 4 main scripts (start, stop, restart and delete), as before.  In the main, only restart should be necessary.

Once we've updated the vm to use node 16 an update should look like this:

```
git pull
npm ci
npm run restart
```

It doesn't guarantee 100% uptime, simply because we're moving files around, but `npm ci` should be pretty fast now, so the disruption should be minimal.

Is that the kind of thing you're after?

**Comment by raisedadead** [comment]
Cool sounds ideal and do you mind updating the "docs" in that case as well. Also, we will probably need a way to update a single app too, including changing/updating/adding env vars for the particular app.

**Comment by ojeytonwilliams** [comment]
Sure thing, I'll get that those fixed up.

**Comment by ojeytonwilliams** [comment]
Okay, it's all done.

**Review by raisedadead — APPROVED** [review]
Taking a leap pf faith.

**Pull request merged**